android/src/main/kotlin/com/rudderstack/sdk/kotlin/android/plugins/devicemode/DeviceModeDestinationPlugin.kt

```diff
@@ -0,0 +1,113 @@
+package com.rudderstack.sdk.kotlin.android.plugins.devicemode
+
+import com.rudderstack.sdk.kotlin.core.Analytics
+import com.rudderstack.sdk.kotlin.core.internals.logger.LoggerAnalytics
+import com.rudderstack.sdk.kotlin.core.internals.models.Event
+import com.rudderstack.sdk.kotlin.core.internals.plugins.Plugin
+import com.rudderstack.sdk.kotlin.core.internals.plugins.PluginChain
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+
+private const val MAX_QUEUE_SIZE = 1000
+
+/*
+ * This plugin will queue the events till the sourceConfig is fetched and
+ * will host all the device mode destination plugins in its PluginChain instance.
+ */
+internal class DeviceModeDestinationPlugin : Plugin {
+
+    override val pluginType: Plugin.PluginType = Plugin.PluginType.Destination
+
+    override lateinit var analytics: Analytics
+
+    private lateinit var pluginChain: PluginChain
+    private val pluginList: MutableList<DestinationPlugin> = mutableListOf()
```

Comment:
Let's use `destination` terminology to signify the actual use case:

```suggestion
    private lateinit var destinationPluginChain: PluginChain
    private val destinationPluginList: MutableList<DestinationPlugin> = mutableListOf()
```